pip_run/scripts.py

```diff
@@ -73,7 +78,35 @@ def search(cls, params):
         return cls.try_read(next(files, None)).params()
 
     def read(self):
-        return self.read_comments() or self.read_python()
+        return self.read_toml() or self.read_comments() or self.read_python()
+
+    def read_toml(self):
+        r"""
+        >>> DepsReader('# /// script\n# dependencies = ["foo", "bar"]\n# ///\n').read()
+        ['foo', 'bar']
+        >>> DepsReader('# /// pyproject\n# dependencies = ["foo", "bar"]\n# ///\n').read_toml()
+        []
+        >>> DepsReader('# /// pyproject\n#dependencies = ["foo", "bar"]\n# ///\n').read_toml()
+        []
+        >>> DepsReader('# /// script\n# dependencies = ["foo", "bar"]\n').read_toml()
+        []
+        """
+        TOML_BLOCK_REGEX = r'(?m)^# /// (?P<type>[a-zA-Z0-9-]+)$\s(?P<content>(^#(| .*)$\s)+)^# ///$'
```

Comment:
I would also suggest moving this out of the local scope to a global constant somewhere in the upper part of the module.

Reply:
I'm inclined not to. Partly because keeping all the code that was directly copied from the PEP together, but also I don't see the benefit. Nothing else uses this value, and referencing it as a local is marginally faster than referencing it as a global. The performance is, of course, utterly irrelevant compared to the cost of reading the file we're parsing, so that's not a compelling argument. But equally, moving it to the top "because that's where global constants go" is also not a compelling argument (to me, anyway).

Reply:
I see your point. My suggestion was not in any way motivated by performance.
This just looks like a constant and [according to PEP 8, constants should be placed after globals, after imports](https://peps.python.org/pep-0008/#imports).
When it comes to performance, it might be improved by compiling the pattern in the module scope, which I didn't suggest for the reasons you've mentioned.

Reply:
If I may put my two cents in, I would make that pattern a global precompiled constant if it were up to me, it just sounds like good and consistent practice to me even though perf is irrelevant.
I personally don't find the "all the code that was directly copied from the PEP together" bit convincing at all 😄

Reply:
Pre-compiling the pattern in the module scope has one underlying function, that was not mentioned: it validates the pattern.

Reply:
I don't particularly want to get into an extended debate here. Ultimately, I'd like to see this merged, and I'm not entirely sure if any of the current participants in this discussion are able to merge a PR here. I'm willing to move `TOML_BLOCK_REGEX` to a global compiled regex, or even just a global string, if @jaraco or another project maintainer feels it would be a useful change. But otherwise I'd prefer just to leave it as it stands. I'm not particularly convinced by "because PEP 8 says so" arguments, TBH.

(Yes, my comment about performance was silly. Sorry.)

Reply:
> I'm willing to move TOML_BLOCK_REGEX to a global compiled regex, or even just a global string, if @jaraco or another project maintainer feels it would be a useful change. But otherwise I'd prefer just to leave it as it stands.

I was assigned to the issue this PR is meant to solve (#95) _after_ your PR had been submitted. That's why I allowed myself to leave a review. But of course, I agree, everything is up to jaraco.